diff --git a/kiss b/kiss
index 8038def..37ebe0d 100755
--- a/kiss
+++ b/kiss
@@ -1497,6 +1497,14 @@ pkg_install() {
     # if needed.
     decompress "$tar_file" | tar xf -
 
+    # FIXME: make sure to use 'as_user', since this command requires root.
+    # Fix a bug when FreeBSD's bsd-base, since some of the files are marked
+    # with a flag that makes them immutable.
+    case "$(uname -s)" in
+        FreeBSD) chflags -R noschg "$PWD";;
+        #Linux) chattr -i -R "$PWD";;
+    esac
+
     # Naively assume that the existence of a manifest file is all that
     # determines a valid KISS package from an invalid one. This should be a
     # fine assumption to make in 99.99% of cases.
